tool/teleport/common/teleport.go

```diff
@@ -990,3 +1023,83 @@ func onIntegrationConfListDatabasesIAM(params config.IntegrationConfListDatabase
 
 	return nil
 }
+
+func onIntegrationBootstrapCreateExternalCloudAuditCmd(params config.IntegrationBootstrapCreateExternalCloudAudit) error {
+	ctx := context.Background()
+	cfg, err := awsConfig.LoadDefaultConfig(ctx, awsConfig.WithRegion(params.Region))
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	if params.WithCMK && params.KMSKeyID != "" {
+		return trace.BadParameter("Specifying a pre-existing kms key is not supported when specifying --with-cmk")
+	}
+
+	err = externalcloudaudit.ValidateBuckets(params.EventBucket, params.SessionBucket, params.QueryResultsBucket, params.LargePayloadBucket)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	uniqueBuckets := map[string]struct{}{}
+	eventBucket, _, err := externalcloudaudit.ParseS3URI(params.EventBucket)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	uniqueBuckets[eventBucket] = struct{}{}
+	sessionBucket, _, err := externalcloudaudit.ParseS3URI(params.SessionBucket)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	uniqueBuckets[sessionBucket] = struct{}{}
+	queryResultBucket, _, err := externalcloudaudit.ParseS3URI(params.QueryResultsBucket)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	uniqueBuckets[queryResultBucket] = struct{}{}
+	largePayloadBucket, _, err := externalcloudaudit.ParseS3URI(params.LargePayloadBucket)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	uniqueBuckets[largePayloadBucket] = struct{}{}
+
+	kmsClient := kms.NewFromConfig(cfg)
+	if params.WithCMK {
+		keyOutput, err := externalcloudaudit.EnsureKMSKey(ctx, kmsClient, externalcloudaudit.EnsureKMSKeyRequest{
+			Alias: params.KMSKeyAlias,
+		})
+		if err != nil {
+			return trace.Wrap(err)
+		}
+
+		params.KMSKeyID = keyOutput.KMSKeyID
```

Comment:
I think this is not idempotent because we will lose the key ID after the run. Maybe we can fetch it by alias, or, do you think we need KMS encryption for the MVP with resource bootstrapping or would the default SSE-S3 encryption be fine?

Reply:
I think for this MVP we can just use default SSE-S3.